pkg/segment/writer/segstore.go

```diff
@@ -908,11 +893,10 @@ func (segstore *SegStore) computeStarTree() {
 			}
 		}
 
-		segstore.sbuilder.ResetSegTree(sortedGrpCols, mCols,
-			segstore.stbDictEncWorkBuf)
+		segstore.stbHolder.stbPtr.ResetSegTree(sortedGrpCols, mCols, segstore.stbDictEncWorkBuf)
 	}
 
-	if !segstore.usingSegTree { // if tree creation had failed on first block, then skip it
+	if segstore.stbHolder == nil { // if tree creation had failed on first block, then skip it
```

Comment:
the comment needs to be updated based on my above feedback

Reply:
Addressed in the latest commit.